src/globus_sdk/services/tunnels/client.py

```diff
@@ -0,0 +1,330 @@
+        :param lifetime_mins: The amount of time in minutes that the Tunnel will
+            run. The default is 360 minutes (6 hours).
+        :param query_params: Any additional parameters will be passed through
+            as query params.
+
+        .. tab-set::
+
+            .. tab-item:: Example Usage
+
+                .. code-block:: python
+
+                    tc = globus_sdk.TunnelClient(...)
+                    result = tc.create_tunnel(listener_sap, initiator_sap)
+                    print(result["data"]["id"])
+
+            .. tab-item:: API Info
+
+                ``POST /tunnels``
+        """
+        if submission_id is None:
+            submission_id = uuid.uuid4()
+
+        attributes: dict[str, t.Any] = {
+            "submission_id": str(submission_id),
+        }
+        if label:
+            attributes["label"] = label
+        if restartable is not None:
+            attributes["restartable"] = restartable
+        if lifetime_mins is not None:
+            attributes["lifetime_mins"] = lifetime_mins
+
+        data = {
+            "data": {
+                "type": "Tunnel",
+                "attributes": attributes,
+                "relationships": {
+                    "listener": {
+                        "data": {
+                            "type": "StreamAccessPoint",
+                            "id": listener_stream_ap_id,
+                        }
+                    },
+                    "initiator": {
+                        "data": {
+                            "type": "StreamAccessPoint",
+                            "id": initiator_stream_ap_id,
+                        }
+                    },
+                },
+            }
+        }
+
+        r = self.post("/v2/tunnels", data=data, query_params=query_params)
+        return r
+
+    def show_tunnel(
+        self,
+        tunnel_id: str,
+        *,
+        query_params: dict[str, t.Any] | None = None,
+    ) -> response.GlobusHTTPResponse:
+        """
+        :param tunnel_id: The ID of the Tunnel which we are fetching details about.
+        :param query_params: Any additional parameters will be passed through
+            as query params.
+
+        .. tab-set::
+
+            .. tab-item:: Example Usage
+
+                .. code-block:: python
+
+                    tc = globus_sdk.TunnelClient(...)
+                    result = tc.show_tunnel(tunnel_id)
+                    print(result["data"])
+
+            .. tab-item:: API Info
+
+                ``GET /tunnels/<tunnel id>``
+        """
+        r = self.get(f"/v2/tunnels/{tunnel_id}", query_params=query_params)
+        return r
+
+    def set_listener(
+        self,
+        tunnel_id: str,
+        listener_cs: str,
+        *,
+        query_params: dict[str, t.Any] | None = None,
+    ) -> response.GlobusHTTPResponse:
+        """
+        :param tunnel_id: The ID of the Tunnel which the application listener's
+            contact string will be set.
```

Comment:
This param string doesn't look correct, but double-check me on that.

```suggestion
        :param tunnel_id: The ID of the Tunnel.
```